### PR TITLE
refactor: extract lookup

### PR DESCRIPTION
**Pull request by jef**
### Description

- Extract lookup process from #110
- Add `await` back to function calls that are `async`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -41,7 +23,7 @@ async function main() {
 
 	for (const store of Stores) {
 		Logger.debug(store.links);
-		setTimeout(tryLookupAndLoop, getSleepTime(), browser, store);
+		setTimeout(await tryLookupAndLoop, getSleepTime(), browser, store);
```
Comment (Doridian) [inline src/index.ts:26]: `await` cannot be used here, at the very least it won't work.

File: src/store/lookup.ts [hunk 2]
```diff
@@ -81,3 +81,20 @@ export async function lookup(browser: Browser, store: Store) {
...
+	Logger.debug(`[${store.name}] Starting lookup...`);
+	try {
+		await lookup(browser, store);
+	} catch (error) {
+		Logger.error(error);
+	}
+
+	const sleepTime = getSleepTime();
+	Logger.debug(`[${store.name}] Lookup done, next one in ${sleepTime} ms`);
+	setTimeout(await tryLookupAndLoop, sleepTime, browser, store);
```
Comment (Doridian) [inline src/store/lookup.ts:99]: See previous comment

File: src/index.ts (hunk 1 above)
Comment (jef) [reply src/index.ts:26]: I thought the same, which I thought was weird. It doesn't complain either way. I wonder if `tsc` throws it out...

Comment (jef) [reply src/index.ts:26]: Oh, the linter catches it though!

**Review by Doridian — APPROVED** [review]

**Pull request merged**